pkg/api/common.go

```diff
@@ -92,14 +92,17 @@ type Config struct {
 	Auth         *Auth
 	MultiTenancy tenancy.Authorizer
 
+	PromscaleEnableFeatures      string
+	PromscaleEnabledFeaturesMap map[string]struct{}
```

Comment:
```suggestion
	EnableFeatures      string
	EnabledFeaturesMap map[string]struct{}
```

I think these should be inside `runner/flags.go` and when creating the router, we can pass map down so that `query.NewEngine` recognizes the `promql-.*` based features.

Reply:
Can you elaborate on why you think that it should be in `runner/flags.go`? I have reworked this and structured it such that we do pass the map down to `query.NewEngine`, thanks for the suggestion!